Implement a new WebKit Contributors Meeting registration process.

jdatapple opened the pull request:
#### 9fd3df9d0352c5ddb796d3381a157bb0e98c9cb0
<pre>
Implement a new WebKit Contributors Meeting registration process.
<a href="https://bugs.webkit.org/show_bug.cgi?id=245798">https://bugs.webkit.org/show_bug.cgi?id=245798</a>

Reviewed by Timothy Hatcher.

* Websites/webkit.org/wp-content/plugins/github-oauth/api.php: Added.
* Websites/webkit.org/wp-content/plugins/github-oauth/oauth.php: Added.
* Websites/webkit.org/wp-content/plugins/github-oauth/plugin.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/admin.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/form.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/plugin.php: Added.
* Websites/webkit.org/wp-content/themes/webkit/page-registration.php: Added.

Canonical link: <a href="https://commits.webkit.org/255004@main">https://commits.webkit.org/255004@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff56e094e58651cc56e28c9b5e54ea58bb1d0274

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90991 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35562 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/21525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/100337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/34062 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/83335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/97125 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96647 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/77774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/83335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/21525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/83335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35141 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/21525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/32939 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/21525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/36719 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/77774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1532 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38647 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/21525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Websites/webkit.org/wp-content/plugins/github-oauth/oauth.php:
```diff
@@ -0,0 +1,142 @@
+            'redirect_uri' => $this->redirect_url
+        ]);
+
+        if ($response === null) {
```
Can drop the `{ }` here.

xeenon commented on Websites/webkit.org/wp-content/plugins/github-oauth/oauth.php:
```diff
@@ -0,0 +1,142 @@
+
+        if ($response === null) {
+            return 'Invalid state';
+        } elseif (isset($response->error_description)) {
```
Can just be a `if` since the previous case is a `return`. Also can drop the `{ }`.

xeenon commented on Websites/webkit.org/wp-content/plugins/meeting-registration/admin.php:
```diff
@@ -0,0 +1,131 @@
+    #new-meeting-form.show {
+        display: inline-block;
+    }
+
```
Nit: Drop the extra new line.

xeenon commented on Websites/webkit.org/wp-content/plugins/meeting-registration/admin.php:
```diff
@@ -0,0 +1,131 @@
+        }
+    });
+
+
```
Nit: Drop the extra new lines.

xeenon commented on Websites/webkit.org/wp-content/plugins/meeting-registration/form.php:
```diff
@@ -0,0 +1,120 @@
+    <div class="alignright">
+        <input type="submit" name="Submit" value="Register" class="submit-button">
+    </div>
+
```
Nit: Drop the extra newline.

jdatapple committed:
Implement a new WebKit Contributors Meeting registration process.
https://bugs.webkit.org/show_bug.cgi?id=245798

Reviewed by Timothy Hatcher.

* Websites/webkit.org/wp-content/plugins/github-oauth/api.php: Added.
* Websites/webkit.org/wp-content/plugins/github-oauth/oauth.php: Added.
* Websites/webkit.org/wp-content/plugins/github-oauth/plugin.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/admin.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/form.php: Added.
* Websites/webkit.org/wp-content/plugins/meeting-registration/plugin.php: Added.
* Websites/webkit.org/wp-content/themes/webkit/page-registration.php: Added.

Canonical link: https://commits.webkit.org/255004@main
